src/_guides/libraries/objective-c-interop.md

```diff
@@ -293,6 +293,58 @@ then check the status, and wait for the duration of the audio file:
     }
 ```
 
+### Callbacks and multithreading limitations
+
+Multithreading issues are the biggest limitation
+of Dart's experimental support for Objective-C interop.
+While `ffigen` supports converting
+Dart functions to Objective-C blocks,
+these limitations must be kept in mind.
+The following issues also affect the use of certain Apple APIs:
+
+* Dart isolates are not the same thing as threads.
+   An isolate isn't guaranteed to run on a particular thread,
+   and the VM might change which thread an isolate is running on
+   without warning.
+   There is an [open feature request][] to allow isolates to be
+   pinned to specific theads.
+* Most Apple APIs don't have any guarantees about
+   on which thread a callback will run.
+* Most APIs that involve UI interaction
+   can only be called on the main thread,
+   also called the _platform_ thread in Flutter.
+4. Many Apple APIs are [not thread safe][].
+
+Points 1 and 2 mean that a callback created in one isolate
+may be invoked on a thread running a different isolate,
+or no isolate at all.
+This will cause your app to crash.
+You can work around this limitation by writing some
+Objective-C code that intercepts your callback and
+forwards it over a [`Dart_Port`]({{site.dart-api}}/dart-ffi/NativePort.html)
+to the correct isolate.
+For an example of this,
+see the implementation of [`package:cupertino_http`][].
+
+The third point means that directly calling some Apple APIs
+using the generated Dart bindings might be thread unsafe.
+This could crash your app, or cause other unpredictable behavior.
+You can work around this limitation by writing some
+Objective-C code that dispatches your call
+to the main thread.
+For more information, see the [Objective-C dispatch documentaion][].
+
+Regarding the last point,
+although Dart isolates can switch threads,
+they only ever run on one thread at a time.
+So, the API you are interacting with
+doesn't necessarily have to be thread safe,
+as long as it is not thread hostile,
```

Comment:
Does "thread hostile" need to be clarified or would readers of this section already understand this term?

Reply:
It might be nice to link to a definition, but I'm not sure what would be official enough.

Reply:
Alternatively, this thread hostility clause is sort of saying the same thing as the thread constraints clause, so maybe we just remove this bit?

Reply:
Reading through this again, I think it's fine, especially with that extra clause. If there's confusion in the future we can revisit the mention. Thanks!